ouroboros-consensus/src/Ouroboros/Consensus/Node/Run.hs

```diff
@@ -98,4 +111,16 @@ class ( LedgerSupportsProtocol           blk
       , ShowProxy                (Header blk)
       , ShowProxy                 (Query blk)
       , ShowProxy           (TxId (GenTx blk))
-      ) => RunNode blk
+      ) => RunNode blk where
+  -- | A protocol may or may not be able to determine from the 'Header' whether
+  -- the header's block was self-issued.
+  --
+  -- TODO Only the @Shelley@ currently returns anything other than
+  -- 'MaybeSelfIssued'.
+  isSelfIssued :: TopLevelConfig blk -> Header blk -> WhetherSelfIssued
```

Comment:
Add a comment on `RunNode`; here's the idea. We want to keep `RunNode` empty. One reason for keeping it empty is so that delegating-HFC instances can have precise contexts: eg `LedgerSupportsPeerSelection` for HF block requires `LedgerSupportsPeerSelection` for each era. It may be right now, but in the past and in the future, `RunNode` and each of its superclasses need not necessarily be so compositional. Summary: we want the HFC instances to be explicitly compositional when possible.

Reply:
Related: Pull this method out into its own `SupportsTracing` class; something that obviously exists because of business-level as opposed to correctness concerns (to the extent that those two are separable 👍).